discretize/TensorMesh.py

```diff
@@ -327,3 +332,45 @@ def cellBoundaryInd(self):
             indzd = (self.gridCC[:, 2] == min(self.gridCC[:, 2]))
             indzu = (self.gridCC[:, 2] == max(self.gridCC[:, 2]))
             return indxd, indxu, indyd, indyu, indzd, indzu
+
+    @property
+    def dim_names(self):
+        """Names of the dimensions.
+
+        Different for different meshes.
+        """
+        return ['x', 'y', 'z'][:self.dim]
+
+    @property
+    def attributes(self):
```

Comment:
I appreciate that this is handy, but I am a bit hesitant at this point to add another property to the Tensor Mesh that contains much of the same information as is already available through other properties (e.g. the list in the docs is pretty extensive already... http://discretize.simpeg.xyz/en/master/api/generated/discretize.TensorMesh.html#module-discretize.TensorMesh) 

What do you think of performing these actions in the print statement?

Reply:
@lheagy, you confuse me here... :smile:

I made this a property because in your previous review (https://github.com/simpeg/discretize/pull/143#pullrequestreview-224688874) you said "this could be a property on the mesh - it would make it easier to inherit (then on the cyl mesh for example, all of this code could be re-used". But I am happy to go back to what it was.

Reply:
Sorry for the confusion! In the previous review, I was only referring to the `dimension_names`, and it was really only after working on #134 that I got a sense of how over-loaded the attributes are.

If you prefer to leave this as a property, I think it is okay, as long as we make it a private variable e.g. `_attributes`